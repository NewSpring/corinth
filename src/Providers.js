import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
