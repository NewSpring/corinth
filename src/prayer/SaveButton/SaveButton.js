import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { ChannelLabel, Button } from '@apollosproject/ui-kit';

const SaveButton = memo(({ saved, onPress }) =>
  saved ? (
    <Button onPress={onPress}>
      <ChannelLabel icon="like-solid" label={'Saved'} />
    </Button>
  ) : (
    <Button onPress={onPress} bordered>
      <ChannelLabel icon="like" label={'Save Prayer'} />
    </Button>
  )
);

SaveButton.propTypes = {
  saved: PropTypes.bool,
  onPress: PropTypes.func,
};

SaveButton.defaultProps = {
  saved: false,
  onPress: () => null,
};

SaveButton.displayName = 'SaveButton';

export default SaveButton;
