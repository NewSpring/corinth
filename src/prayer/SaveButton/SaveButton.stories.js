import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { PaddedView } from '@apollosproject/ui-kit';

import SaveButton from '.';

storiesOf('prayer/SaveButton', module).add('default', () =>
  (
    <PaddedView>
      <SaveButton />
    </PaddedView>
  ).add('Saved', () => (
    <PaddedView>
      <SaveButton saved />
    </PaddedView>
  ))
);
