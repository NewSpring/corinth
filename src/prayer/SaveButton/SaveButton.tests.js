import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import SaveButton from '.';

describe('the SaveButton component', () => {
  it('renders a SaveButton', () => {
    const tree = renderer.create(
      <Providers>
        <SaveButton />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders a saved SaveButton', () => {
    const tree = renderer.create(
      <Providers>
        <SaveButton saved />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
