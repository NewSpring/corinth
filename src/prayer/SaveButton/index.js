import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
import SAVE_PRAYER from '../data/mutations/savePrayer';
import UNSAVE_PRAYER from '../data/mutations/unSavePrayer';
import GET_SAVED_PRAYERS from '../data/queries/getSavedPrayers';
import SaveButton from './SaveButton';

const SaveButtonConnected = memo(({ prayerID, saved, toggleSavedState }) => (
  <Mutation mutation={SAVE_PRAYER}>
    {(save) => (
      <Mutation mutation={UNSAVE_PRAYER}>
        {(unSave) => (
          <AnalyticsConsumer>
            {({ track }) => (
              <SaveButton
                saved={saved}
                onPress={() => {
                  toggleSavedState();
                  if (saved)
                    unSave({
                      variables: { nodeId: prayerID },
                      refetchQueries: [{ query: GET_SAVED_PRAYERS }],
                    });
                  else
                    save({
                      variables: { nodeId: prayerID },
                      refetchQueries: [{ query: GET_SAVED_PRAYERS }],
                    });
                  track({
                    eventName: saved ? 'Unsaved Prayer' : 'Saved Prayer',
                  });
                }}
              />
            )}
          </AnalyticsConsumer>
        )}
      </Mutation>
    )}
  </Mutation>
));

SaveButtonConnected.propTypes = {
  prayerID: PropTypes.string,
  saved: PropTypes.bool,
  toggleSavedState: PropTypes.func,
};

SaveButtonConnected.displayName = 'SaveButtonConnected';

export default SaveButtonConnected;
