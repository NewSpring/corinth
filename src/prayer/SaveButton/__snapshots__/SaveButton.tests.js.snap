// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SaveButton component renders a SaveButton 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    bordered={true}
    disabled={false}
    pill={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#6bac43",
        "borderRadius": 48,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "center",
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      flexed={false}
      style={
        Object {
          "alignItems": "center",
          "flex": null,
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={19.2}
        bbWidth={19.2}
        height={19.2}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            null,
            Object {
              "flex": 0,
              "height": 19,
              "width": 19,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={19.2}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
            fill={
              Array [
                0,
                4285246531,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "paddingHorizontal": 4,
          }
        }
        withIcon="like"
      >
        <Text
          style={
            Object {
              "color": "#6bac43",
              "fontFamily": "Colfax-Bold",
              "fontSize": 12,
              "lineHeight": 17.28,
            }
          }
          tint="#6bac43"
        >
          Save Prayer
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`the SaveButton component renders a saved SaveButton 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    bordered={false}
    disabled={false}
    pill={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6bac43",
        "borderColor": "#6bac43",
        "borderRadius": 48,
        "borderWidth": 2,
        "elevation": 2,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "center",
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      flexed={false}
      style={
        Object {
          "alignItems": "center",
          "flex": null,
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={19.2}
        bbWidth={19.2}
        height={19.2}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            null,
            Object {
              "flex": 0,
              "height": 19,
              "width": 19,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={19.2}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
            fill={
              Array [
                0,
                4294967295,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "paddingHorizontal": 4,
          }
        }
        withIcon="like-solid"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "Colfax-Bold",
              "fontSize": 12,
              "lineHeight": 17.28,
            }
          }
          tint="#ffffff"
        >
          Saved
        </Text>
      </View>
    </View>
  </View>
</View>
`;
